✨ add question page for quiz

mewiemers opened the pull request:
- build buttons in question page
- connect data from db with quizbutton
- give an alert when answer is correct at `onClick`
- change font in button
- add states to change color in button when clicked
- try to set another function with additional text when clicked(is working in progress)

mewiemers committed:
✨ add question page for quiz

- build buttons in question page
- connect data from db with quizbutton
-give an alert when answer is correct at `onClick`

mewiemers committed:
add handleClick to change Color if clicked

lmachens commented on components/quiz/Quiz.tsx:
```diff
@@ -12,10 +13,25 @@ export type Answer = {
 function Quiz({ question, answerOptions }: QuizProps) {
+  const handleAnswerClick = (isCorrect: boolean) => {
+    if (isCorrect === true) {
+      alert("AMAZING"), setColorchange("green");
```
This is an uncommon code style.

```suggestion
      alert("AMAZING");
      setColorchange("green");
```

lmachens commented on components/quiz/Quiz.tsx:
```diff
@@ -12,10 +13,25 @@ export type Answer = {
+      alert("Hmm, try again!"), setColorchange("red");
+    }
+  };
+  const [colorchange, setColorchange] = useState("blue");
```
It's possible to define the state in this line, but for a consistent code style, I would move states and other hooks to the top of this component/function.

lmachens replied:
`camelCase` would be `colorChange`

lmachens replied:
Instead of using colors in this state, I would prefer a more meaningful value like:

```ts
const [guessStatus, setGuessStatus] = useState<'pending' | 'wrong' | 'correct'>('pending');
```

Feel free to outsource the union type:
```ts
type AnswerStatus = 'pending' | 'wrong' | 'correct';
```

Color the button `green` if it is correct is a styling question. Maybe you like to change the style in the future and use `lightgreen` or `#2ea043` as color. This shouldn't require a (js) code change. Instead, I would create CSS classes for `.wrong` and `.correct` answers, which colorize your buttons.

mewiemers replied:
> Instead of using colors in this state, I would prefer a more meaningful value like:
> 
> ```ts
> const [guessStatus, setGuessStatus] = useState<'pending' | 'wrong' | 'correct'>('pending');
> ```
> 
> Feel free to outsource the union type:
> 
> ```ts
> type AnswerStatus = 'pending' | 'wrong' | 'correct';
> ```
> 
> Color the button `green` if it is correct is a styling question. Maybe you like to change the style in the future and use `lightgreen` or `#2ea043` as color. This shouldn't require a (js) code change. Instead, I would create CSS classes for `.wrong` and `.correct` answers, which colorize your buttons.

classNames are set to Quizbutton in CSS, I would like that every button is united with these CSS class and the boolean value, but don't know how I write right CSS  statement in if statement.

lmachens replied:
Take a look at your button component or use a lib like
https://www.npmjs.com/package/classnames

lmachens commented on pages/question.tsx:
```diff
@@ -0,0 +1,24 @@
+  useEffect(() => {
+    fetch("/api/quizanswers")
+      .then((response) => response.json())
+      .then((response) => setChoice(response[0]));
```
```suggestion
      .then((result) => setChoice(result[0]));
```
or
```suggestion
      .then((quizAnswers) => setChoice(quizAnswers[0]));
```

lmachens replied:
Why is it required to read only the first answer?

mewiemers replied:
> Why is it required to read only the first answer?

don't understand what you meant, "read-only the first answer"?

lmachens replied:
Because you access `[0]`. Why do you send an array from your api if you only need the first entry?

lmachens commented on pages/question.tsx:
```diff
@@ -0,0 +1,24 @@
+  }
+
+  return (
+    <>
+      <Quiz answerOptions={choice.answerOptions} question={choice.question} />
+    </>
```
```suggestion
    <Quiz answerOptions={choice.answerOptions} question={choice.question} />
```

lmachens reviewed (commented):
gj 🚀

mewiemers committed:
✨ add function for changing button color at `onClick`

- change font in button
- add states to change color in button when clicked
- try to set another function with additional text when clicked(is working in progress)

mewiemers committed:
add type

DNaunin reviewed (approved):
Great Job! 🎉

mewiemers committed:
🏷️ refactor types
